src/SourceLocator/Ast/Parser/MemoizingParser.php

```diff
@@ -0,0 +1,44 @@
+<?php
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\SourceLocator\Ast\Parser;
+
+use PhpParser\ErrorHandler;
+use PhpParser\Node;
+use PhpParser\Parser;
+
+/**
+ * @internal
+ */
+final class MemoizingParser implements Parser
```

Comment:
TODO: test this

Reply:
Tested

Reply:
Is that intentional name or typo in `Memorizing`?

Reply:
https://en.m.wikipedia.org/wiki/Memoization

Reply:
Oh, thanks for new term